build.rs

```diff
@@ -1,11 +1,29 @@
 extern crate cc;
+use std::env;
+
+const DEPLOYMENT_TARGET_VAR: &str = "MACOSX_DEPLOYMENT_TARGET";
 
 fn main() {
     if cfg!(target_os = "macos") {
+        let min_version = match env::var(DEPLOYMENT_TARGET_VAR) {
+            Ok(ver) => ver,
+            Err(_) => String::from(match env::var("CARGO_CFG_TARGET_ARCH").unwrap().as_str() {
+                "x86_64" => "10.8", // NSUserNotificationCenter first showed up here.
+                "aarch64" => "11.0", // Apple silicon started here.
```

Comment:
Optional: Unless you know something I don't, I would skip the aarch64 default and just go with 10.8 for both. I imagine Apple has already thought about this and have mitigations in place. Why? One less input variable to worry about, to simplify future hairy debugging issues like this one.

Reply:
I can add a link to it, but this is also what rustc's Apple base target definition uses (I would have copy pasted the x86 one too but 10.7 is too old). M1/M2 chips can't run on anything older than 11.

Reply:
Ah, thanks for the context. Doing the same as rustc seems better. Thanks.